### PR TITLE
Add ERC721ALowCap + tests

### PR DESCRIPTION
**Pull request by Austinhs**
Ended up creating a function "tokensOfOwner" as its something Divine Anarchy uses a lot within its dApp experience. Figured I would create a pull request incase it was something you guys wanted to add to the main repo :)

### CODE REVIEW

**Comment by Vectorized** [comment]
See https://github.com/chiru-labs/ERC721A/issues/115 on `tokensOfOwner` function.

**Comment by Vectorized** [comment]
I suggest putting the README update into another PR, so that the maintainers can merge that first.

**Comment by Austinhs** [comment]
@Vectorized in #115 you compressed the function by a bit

```
    /**
     * @dev Returns the tokenIds of the address. O(totalSupply) in complexity.
     */
    function tokensOfOwner(address owner) public view returns (uint256[] memory) {
        uint256 holdingAmount = balanceOf(owner);
        uint256 currSupply    = _currentIndex;
        uint256 tokenIdsIdx;
        address currOwnershipAddr;

        uint256[] memory list = new uint256[](holdingAmount);

        unchecked {
            for (uint256 i; i < currSupply; i++) {
                TokenOwnership memory ownership = _ownerships[i];

                if (ownership.burned) {
                    continue;
                }

                // Find out who owns this sequence
                if (ownership.addr != address(0)) {
                    currOwnershipAddr = ownership.addr;
                }

                // Append tokens the last found owner owns in the sequence
                if (currOwnershipAddr == owner) {
                    list[tokenIdsIdx++] = i;
                }

                // All tokens have been found, we don't need to keep searching
                if(tokenIdsIdx == holdingAmount) {
                    break;
                }
            }
        }

        return list;
    }
```

Specifically you removed this:
```

                // All tokens have been found, we don't need to keep searching
                if(tokenIdsIdx == holdingAmount) {
                    break;
                }
```

Am I wrong in assuming this loop is going to continue? Or is it because this is a view it does not matter and its better to save the space in the contract to lower deploy cost. (I think things like comments, spacing etc will get removed with the optimizer but I could be wrong)

**Comment by Vectorized** [comment]
@Austinhs You can add the break if you want. I simply copied and pasted from my own code. I didn't code the break it cuz its just a view function.

File: contracts/extensions/ERC721ALowCap.sol
Comment (ahbanavi) [inline contracts/extensions/ERC721ALowCap.sol:19]: Remove white spaces before the assignment.

File: contracts/mocks/ERC721ALowCap.sol
Comment (ahbanavi) [inline contracts/mocks/ERC721ALowCap.sol:8]: File name better be the same as contract name, try changing the file name to `ERC721ALowCapMock.sol`.

File: test/extensions/ERC721ALowCap.test.js
Comment (ahbanavi) [inline test/extensions/ERC721ALowCap.test.js:10]: Use 2 spaces for indentation in this context instead of 4.

**Review by ahbanavi — CHANGES_REQUESTED** [review]
Some style notes.
Also, add some tests for burned tokens too, just like transfer.
And it would be better if you add tests with `ERC721AOwnersExplicit` too.

File: test/extensions/ERC721ALowCap.test.js
Comment (johnnyshankman) [reply test/extensions/ERC721ALowCap.test.js:10]: Should be able to run `npm run lint` to autofix all formatting issues!

**Comment by Austinhs** [comment]
Fixed the linting errors. Thanks for the `npm run lint` suggestion @johnnyshankman definitely saved a few mins of my time <3.

@ahbanavi , you mentioned adding tests with `ERC721AOwnersExplicit` did you just want it to be included in the mock or did you have something else in mind?

Sorry for the holdup on this branch, its been hectic month or so and just recently getting back to work on contract stuff again so I figured I'd finish this up as well as I knew it was still open.

**Comment by Vectorized** [comment]
I'll need to check with the team if they are ok with this extension. 

We may consider a different name if we decide to go ahead with it.

**Comment by ahbanavi** [comment]
@Austinhs glad to have you back
A lot has changed since you opened this PR. 
We should change the base extension to work with `_startTokenId` and use `createTestSuite` in tests.  
I work on this on my fork and open a PR to your branch; tests for `ERC721AOwnersExplicit` can wait.

**Review by ahbanavi — APPROVED** [review]
IMO it's good to go 
@Vectorized

**Comment by Vectorized** [comment]
Lemme sit on it for a while, probably will merge this weekend if team agrees.

**Comment by Vectorized** [comment]
I've made a new PR that is based on this. 

This (#114) will be merged first if #224 is approved.

**Pull request merged**
